dev-tools/mage/settings.go

```diff
@@ -561,13 +562,80 @@ var (
 // import path and the current directory's import path.
 func GetProjectRepoInfo() (*ProjectRepoInfo, error) {
 	repoInfoOnce.Do(func() {
-		repoInfoValue, repoInfoErr = getProjectRepoInfo()
+		if isUnderGOPATH() {
+			repoInfoValue, repoInfoErr = getProjectRepoInfoUnderGopath()
+		} else {
+			repoInfoValue, repoInfoErr = getProjectRepoInfoWithModules()
+		}
 	})
 
 	return repoInfoValue, repoInfoErr
 }
 
-func getProjectRepoInfo() (*ProjectRepoInfo, error) {
+func isUnderGOPATH() bool {
+	rel, err := filepath.Rel(build.Default.GOPATH, CWD())
+	if err != nil {
+		return false
+	}
+
+	if strings.Contains(rel, "..") {
+		return false
+	}
+
+	return true
+}
+
+func getProjectRepoInfoWithModules() (*ProjectRepoInfo, error) {
+	var (
+		cwd            = CWD()
+		rootImportPath string
+		rootDir        string
+		subDir         string
+	)
+
+	possibleRoot := cwd
+	var errs []string
+	for {
+		isRoot, err := isRepoRoot(possibleRoot)
+		if err != nil {
+			errs = append(errs, err.Error())
+		}
+
+		if isRoot {
+			rootDir = possibleRoot
+			break
+		}
+
+		subDir = path.Base(possibleRoot)
+		possibleRoot = path.Dir(possibleRoot)
+	}
+
+	rootImportPath = "github.com/elastic/beats"
```

Comment:
This value needs to come from the go.mod file because you can use any module path you want there. There is no direct relation between the file path and root module path/root import path with go.mod.

Reply:
You are right about the lack of relation between file path and root import path in general. This is why this value should not come from `go.mod` in our case. The module name of the repo is going to be `github.com/elastic/beats/v7`. But still the repo is going to be placed under `github.com/elastic/beats`. If I understood correctly, `RootImportPath` is used when decinding the mount point of the crossBuilder: https://github.com/elastic/beats/blob/master/dev-tools/mage/crossbuild.go#L218

Reply:
Hm, I see it is used for prefixing the vendored libs as well... Then it needs a bit more work.